compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2282,7 +2357,7 @@ pub struct TargetOptions {
     /// Note that the support here is at a codegen level. If the machine code with sanitizer
     /// enabled can generated on this target, but the necessary supporting libraries are not
     /// distributed with the target, the sanitizer should still appear in this list for the target.
-    pub supported_sanitizers: SanitizerSet,
+    pub supported_sanitizers: SanitizerSupport,
```

Comment:
I think it's preferable to avoid breaking target spec users without a strong necessity.
There can be two fields here
```rust
pub stable_sanitizers: SanitizerSet,
pub supported_sanitizers: SanitizerSet,
```
"Stable" takes priority over "supported" if some sanitizer happens to be duplicated in both lists.
(There also will be less churn to the target specs this way.)

Reply:
And the same for compiletest.
```rust
    #[serde(rename = "stable-sanitizers", default)]
    pub(crate) stable_sanitizers: Vec<Sanitizer>,
    #[serde(rename = "supported-sanitizers", default)]
    pub(crate) supported_sanitizers: Vec<Sanitizer>,
```

Reply:
I'm wondering whether the costs--these don't change much after all--outweigh the benefits long term. Unlike explicitly having stable vs unstable under supported_sanitizers, stable_sanitizers vs supported_sanitizers and duplicated entries seem ambiguous and may some cause confusion later. What do you think?

Reply:
I think it's fine long term too.
Right now target specs are mostly simple key-value lists, there are many opportunities to introduce some deeper structures in them besides sanitizer options, but it's probably easier for everyone to just keep (almost) everything flat.

Reply:
Sounds good then! I'll make those changes.

Reply:
The internal `struct SanitizerSupport` can also be removed, and changes to many built-in targets (without stable sanitizers) can be avoided.

Reply:
Done.